Revit_Core_Engine/Query/DoesIntersect.cs

```diff
@@ -0,0 +1,170 @@
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Check if two elements intersect.")]
+        [Input("element1", "First element to check the intersection for.")]
+        [Input("element2", "Second element to check the intersection for.")]
+        [Output("bool", "Result of the intersect checking.")]
+        public static bool DoesIntersect(this Element element1, Element element2)
+        {
+            if (element1 == null || element2 == null)
+            {
+                return false;
+            }
+
+            Transform transform1 = element1.Document.LinkTransform();
+            Transform transform2 = element2.Document.LinkTransform();
+
+            //host vs host
+            if (transform1 == null && transform2 == null)
+            {
+                ElementIntersectsElementFilter intersectFilter = new ElementIntersectsElementFilter(element2);
+                return intersectFilter.PassesFilter(element1);
+            }
+            //host vs link || link vs host
+            else if ((transform1 == null && transform2 != null) || (transform1 != null && transform2 == null))
+            {
+                return DoesIntersectHostVsLink(element1, transform1, element2, transform2);
+            }
+            //link vs link
+            else if (transform1 != null && transform2 != null)
+            {
+                if (transform1.AlmostEqual(transform2))
+                {
+                    ElementIntersectsElementFilter intersectFilter = new ElementIntersectsElementFilter(element2);
+                    return intersectFilter.PassesFilter(element1);
+                }
+
+                List<Solid> element2Solids = element2.Solids(new Options());
+                Transform doubleTransform = transform2.Multiply(transform1.Inverse);
+                List<Solid> el2TransSolids = element2Solids.Select(x => SolidUtils.CreateTransformed(x, doubleTransform)).ToList();
+
+                foreach (Solid el2Solid in el2TransSolids)
+                {
+                    ElementIntersectsSolidFilter intersectFilter = new ElementIntersectsSolidFilter(el2Solid);
+                    if (intersectFilter.PassesFilter(element1))
+                        return true;
+                }
+            }
+
+            return false;
+        }
+
+        /***************************************************/
+
+        [Description("Check if bounding box intersects with element.")]
+        [Input("bbox", "Bounding box to check the intersection for.")]
+        [Input("element", "Element to check the intersection for.")]
+        [Output("bool", "Result of the intersect checking.")]
+        public static bool DoesIntersect(this BoundingBoxXYZ bbox, Element element)
+        {
+            if (bbox == null || element == null)
+            {
+                return false;
+            }
+
+            if (!bbox.Transform.IsIdentity)
```

Comment:
I guess this could be relaxed by allowing translation, so `bbox.Transform.IsTranslation` and then adding the origin to the min/max of the outline. Or would that be problematic by any means?

Reply:
I think you need to change this line too (please make sure that identity transform returns `IsTranslation == true`).
```suggestion
            if (!bbox.Transform.IsTranslation)
```

Reply:
Yes, my mistake here. Thanks! 👍